tests/pipelines/main.nf.test

```diff
@@ -427,4 +427,79 @@ nextflow_pipeline {
             assert iridanext_metadata.isEmpty()
         }
     }
+
+    test("Testing mismatched IDs") {
+        // IDs in the sample sheet and IDs in the individual MLST JSON files will not match.
+        // This tests the pipelines ability to handle and correct for this problem.
+
+        tag "mismatch"
+
+        when {
+            params {
+                input = "$baseDir/tests/data/samplesheets/samplesheet-mismatched-ids.csv"
+                outdir = "results"
+
+                pd_distm = "scaled"
+                gm_thresholds = "1,0.5,0"
+            }
+        }
+
+        then {
```

Comment:
Could you add a comparison/check in this test related to the `--output-error` file from input_assure? This should contain messages related to all the sample names that were changed to match the identifiers in the samplesheet.csv.

Could you then also add a check for this same file in one of the other test cases where no sample names are changed and verify that no messages are printed in the `--output-error` file?

Could you also then add another test case when some sample names get changed and some do not (that is, some sample identifiers in the samplesheet.csv match those in the mlst.json file, and some do not)?

Reply:
Updated the tests in 34535b98b21f71632612e6667093fdff286d5730